### DISCUSSION TITLE
Add selected Node

### DISCUSSION THREAD

**Discussion by PhilGarb** [xyflow/xyflow#1986] [Q&A] [answered]
When controlling the nodes it does not seem to be possible to add a node and select it right away.

How would I go about adding a selected Node?

**Comment by schadocalex** [comment, marked as answer]
You can add a `selected` property set to `true`.

Note that this is currently undocumented. See https://github.com/wbkd/react-flow/blob/3548891b8e4f6ffbb9de1fbbef582d8eb55304c7/src/types/nodes.ts#L18

but it should work as it is used by the native applyNodeChanges : https://github.com/wbkd/react-flow/blob/316044cd50eb025cd3f757c52a0a216610b619a7/src/utils/changes.ts#L62

**Reply by PhilGarb** [reply]
Thank you for your help! This does seem to work.

**Comment by markwatabe** [comment]
I was going through the documentation trying to find this. It would be great if in the migration docs this were addressed. It was totally unclear to me how I was supposed to replicate this functionality in V10:

`useStoreActions((actions) => actions.setSelectedElements);`
